### PR TITLE
Updated 101 databricks all in one template vnetga

### PR DESCRIPTION
**Pull request by ravrams**
Azure Databricks VNet injection GA changes:

### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Removed security rules for NSG.
* Delegating subnets to 'Microsoft.Databricks/workspaces' service by setting delegations properties.
* Updated api-version to 2019-06-01 which contains all recent properties required for VNet injection GA

### CODE REVIEW

**Comment by ravrams** [comment]
@bmoore-msft  Please do not complete the PR until tomorrow morning.

File: 101-databricks-all-in-one-template-for-vnet-injection/azuredeploy.json [hunk 1]
```diff
@@ -285,7 +111,13 @@
               "addressPrefix": "[parameters('publicSubnetCidr')]",
               "networkSecurityGroup": {
                 "id": "[variables('nsgId')]"
-              }
+              },
+              "delegations": [{
+                "name": "[concat('databricks-del-', uniqueString(parameters('publicSubnetName')))]",
```
Comment (bmoore-msft) [inline 101-databricks-all-in-one-template-for-vnet-injection/azuredeploy.json]: does this need to be globally unique?  if not use a more meaningful name (e.g. databrick-del-public)

File: 101-databricks-all-in-one-template-for-vnet-injection/azuredeploy.json [hunk 2]
```diff
@@ -285,7 +111,13 @@
               "addressPrefix": "[parameters('publicSubnetCidr')]",
               "networkSecurityGroup": {
                 "id": "[variables('nsgId')]"
-              }
+              },
+              "delegations": [{
```
Comment (bmoore-msft) [inline 101-databricks-all-in-one-template-for-vnet-injection/azuredeploy.json]: format the json (e.g. SHIFT+ALT+F in VS Code)

File: 101-databricks-all-in-one-template-for-vnet-injection/azuredeploy.json (hunk 1 above)
Comment (ravrams) [reply 101-databricks-all-in-one-template-for-vnet-injection/azuredeploy.json]: It doesn't needs to be globally unique. It needs to be unique for subnet. So, "databrick-del-public" makes sense. I will update.

**Pull request merged**